crypto/engine/eng_table.c

```diff
@@ -161,7 +163,9 @@ IMPLEMENT_LHASH_DOALL_ARG(ENGINE_PILE, ENGINE);
 
 void engine_table_unregister(ENGINE_TABLE **table, ENGINE *e)
 {
-    CRYPTO_THREAD_write_lock(global_engine_lock);
+    if (!CRYPTO_THREAD_write_lock(global_engine_lock))
```

Comment:
This is very unfortunate.. Same with the next one.

Reply:
Maybe assert here also?

Reply:
there are several places where locks are done inside void functions.  I stopped  adding comments.  There's an open issue #14239.  Should I put the  comment everywhere so they will stick out?  A TODO(void) comment?

As for the assert, it could be here or perhaps pushed down to the actual locking code as discussed below.

Reply:
A TODO is fine for me.

Reply:
created #14321.